SU2_CFD/src/integration_structure.cpp

```diff
@@ -311,8 +311,9 @@ void CIntegration::Adjoint_Setup(CGeometry ****geometry, CSolver *****solver_con
       if (iMGLevel != config[iZone]->GetnMGLevels()) {
         SetRestricted_Solution(RUNTIME_FLOW_SYS, solver_container[iZone][INST_0][iMGLevel][FLOW_SOL], solver_container[iZone][INST_0][iMGLevel+1][FLOW_SOL],
                                geometry[iZone][INST_0][iMGLevel], geometry[iZone][INST_0][iMGLevel+1], config[iZone]);
-        SetRestricted_Gradient(RUNTIME_FLOW_SYS, solver_container[iZone][INST_0][iMGLevel][FLOW_SOL], solver_container[iZone][INST_0][iMGLevel+1][FLOW_SOL],
-                               geometry[iZone][INST_0][iMGLevel], geometry[iZone][INST_0][iMGLevel+1], config[iZone]);
+//        ToDo: The flow solvers do not use the conservative variable gradients
```

Comment:
here

Reply:
Looks like this is outdated. If anything, it should be a restriction of the primitive gradient to make sure everything is well set in multigrid before the adjoint iteration.

I think it is worth trying to remove and see if the continuous adjoint cases fail in the regressions. If so, we can look into it further, since it might need to be adjusted.

Reply:
On the coarse grids the primitive gradients are never computed, they are only used for viscous residuals but they are zero at that point.
The continuous adjoint cases all pass without calling `SetRestricted_Gradient`